src/main/java/org/folio/service/request/RequestBatchResourceService.java

```diff
@@ -0,0 +1,113 @@
+package org.folio.service.request;
+
+import static org.folio.rest.impl.RequestsAPI.REQUEST_TABLE;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.folio.rest.jaxrs.model.Request;
+import org.folio.rest.persist.PostgresClient;
+import org.folio.service.BatchResourceService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.ext.sql.SQLConnection;
+import io.vertx.ext.sql.UpdateResult;
+
+public class RequestBatchResourceService {
+  private static final Logger LOG = LoggerFactory.getLogger(RequestBatchResourceService.class);
+  private static final String REMOVE_POSITIONS_SQL =
+    "UPDATE %s.%s SET jsonb = jsonb - 'position' WHERE id::text IN (%s)";
+
+  private final BatchResourceService batchResourceService;
+  private final String tenantName;
+
+  public RequestBatchResourceService(String tenantName,
+                                     BatchResourceService batchResourceService) {
+
+    this.batchResourceService = batchResourceService;
+    this.tenantName = tenantName;
+  }
+
+  /**
+   * This method executes batch update for the request table.
+   * The batch package must include request copies with removed positions
+   * as a first part of transaction in order to pass 'itemId - position' constraint.
+   * <p>
+   * Example:
+   * Let's say we have following requests in the batch:
+   * 1. Request A.
+   * 2. Request B.
+   * 3. Request C.
+   * <p>
+   * Then actual batch package, that will be executed, is following:
+   * 1. Copy of Request A with position = null.
+   * 1. Copy of Request B with position = null.
+   * 1. Copy of Request C with position = null.
+   * 4. Request A.
+   * 5. Request B.
+   * 6. Request C.
+   *
+   * @param requests        - List of requests to execute in batch.
+   * @param onFinishHandler - Callback function.
+   */
+  public void executeRequestBatchUpdate(
+    List<Request> requests, Handler<AsyncResult<Void>> onFinishHandler) {
+
+    LOG.debug("Removing positions for all request to go through positions constraint");
+    List<Function<SQLConnection, Future<UpdateResult>>> allBatches = new ArrayList<>();
```

Comment:
Are these really batches, or are they a sequence of SQL commands?

Reply:
By the time the request updates are added to this list, are they actually SQL commands?

Reply:
Maybe we need a different name, e.g. database operations?

Reply:
> By the time the request updates are added to this list, are they actually SQL commands?

No, I use `update` method from `PostgresClient` class for update a request.

> Maybe we need a different name, e.g. all database operations?

Works for me.